File: src/main/java/project/rpg/effect/Poison.java
package project.rpg.effect;

import org.bukkit.entity.LivingEntity;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;

public class Poison extends EffectBase {

    @Override
    public void effect() {
        this._entity.addPotionEffect(new PotionEffect(PotionEffectType.POISON, 20*_second, 2,true));
    }

    public Poison(LivingEntity entity, int second) {
        super(entity, second);
    }

}

File: src/main/java/project/rpg/effects/objects/Poison.kt
package project.rpg.effects.objects

import org.bukkit.entity.LivingEntity
import org.bukkit.potion.PotionEffect
import org.bukkit.potion.PotionEffectType
import project.rpg.effects.EffectBase

class Poison(entity: LivingEntity, second: Int) : EffectBase(entity, second) {
    override fun effect(durationMillis: Long) {
        entity.addPotionEffect(PotionEffect(PotionEffectType.POISON, 20 * second, 2, true))
    }
}
